calculator.rb

```diff
@@ -0,0 +1,131 @@
+# Ada C12 - Tiffany Chio
+# This is the long way of making Ruby perform as a calculator.
+# The program prompts for an operator and two operands.
+# Then prints the answer.
+
+# Validation function for the operator
+# The user input must match one of pre-defined keywords/symbols.
+# Will print error message and prompt for a new input until a valid one is obtained.
+def val_operation(operation)
+  valid = false
+  while valid == false
+    if ["add", "+", "subtract", "-", "multiply", "*", "divide", "/", "exponent", "^", "modulo", "%"].include? operation
+      valid = true
+    else
+      puts ""
+      puts "That is not a valid operator. Please try again."
+      print "Please choose one operator(name or symbol): "
+      operation = gets.chomp.downcase
+    end
+  end
+  return operation
+end
+
+# Operand validation function.
+# Checking that the input is a valid number.
+# Will print error message and prompt for a new input until a valid one is obtained.
+# Will also return either int or float as appropriate.
+def val_number(number, prompt)
+  valid = false
+
+  # Function determines if the user entered an int or float by checking
+  # for the prescence of a decimal point.
+  # Converts string input into appropriate class and returns that value.
+  # If...the number is a float
+  if number.match?(/\.+/)
+    while valid == false
+      converted = number.to_f.to_s
+      if converted == number
+        valid = true
+      else
+        puts ""
+        puts "That is not a number. Please try again."
+        print prompt
+        number = gets.chomp
+      end
+    end
+    return number.to_f
+    # Else...the number is an int
+  else
+    while valid == false
+      converted = number.to_i.to_s
+      if converted == number
+        valid = true
+      else
+        puts ""
+        puts "That is not a number. Please try again."
+        print prompt
+        number = gets.chomp
+      end
+    end
+    return number.to_i
+  end
+end
+
+puts `clear`
+
+# Gives users options to select from.
+puts "Welcome to the Calculator program! Which operator would you like to choose?"
+puts "1. add(+)"
+puts "2. subtract(-)"
+puts "3. multiply(*)"
+puts "4. divide(/)"
+puts "5. exponent(^)"
+puts "6. modulo(%)"
+
+puts ""
+
+# Collects user input for operator.
+# User input goes through operator validation function.
+print "Please choose one operator(name or symbol): "
+operation = gets.chomp.downcase
+operation = val_operation(operation)
+
+puts ""
+
+# Collects user input for the first operand.
```

Comment:
You've written almost exactly the same code here twice, to get the first number and the second number. Could you DRY that up by putting this logic in a method?